ament_cmake_pytest/cmake/ament_add_pytest_test.cmake

```diff
@@ -98,6 +98,46 @@ function(ament_add_pytest_test testname path)
     list(APPEND cmd "-We")
   endif()
 
+  option(AMENT_CMAKE_PYTEST_WITH_COVERAGE
+    "Generate coverage information for Python tests"
+    OFF)
+
+  # enable pytest coverage automatically if the package test_depends on pytest_cov
+  if("python3-pytest-cov" IN_LIST ${PROJECT_NAME}_TEST_DEPENDS)
+    set(AMENT_CMAKE_PYTEST_WITH_COVERAGE ON)
+  endif()
+
+  if(AMENT_CMAKE_PYTEST_WITH_COVERAGE)
+    # check if pytest-cov is available
+    ament_has_pytest_cov(has_pytest_cov pytest_cov_version
+      QUIET PYTHON_EXECUTABLE "${ARG_PYTHON_EXECUTABLE}"
+    )
+    if(NOT has_pytest_cov)
+      message(WARNING
+        "The Python module 'pytest-cov' was not found, test coverage will not be produced "
+        "(e.g. on Ubuntu/Debian install the package 'python3-pytest-cov')")
+      return()
+    else()
```

Comment:
Nitpick: since the `if` block returns please change this `else` into an `endif` to avoid indenting the whole current `else` block.

Reply:
Fixed in 5756cf35ff200a2744f979a8fe47416d1db29c3c.

Reply:
Actually, now that I'm looking at it again, it shouldn't return. Unlike `pytest`, if `pytest-cov` isn't found, it should just print that warning and keep going but without coverage. I've fixed this in ab6cce4d13822877b716ee096e5cf21f1f8e459c. Let me know if you do not agree with this.